metrics/coverage_high_water_mark

```diff
@@ -1 +1 @@
-93.6900
+93.7300
```

Comment:
I tested the bulk of the new code added, and threw in some unrelated tests as well to get this number up.